diff --git a/Config/Memcache.php b/Config/Memcache.php
index 9137890..c14eaf4 100644
--- a/Config/Memcache.php
+++ b/Config/Memcache.php
@@ -1,4 +1,36 @@
 <?php
+
+defined('MOODLE_INTERNAL') or die("This can only be accessed through Moodle");
+
+require_once($CFG->dirroot.'/local/memcachedadmin/env.php');
+require_once($CFG->dirroot.'/cache/locallib.php');
+
+global $PHPMEMCACHEDADMIN_DIR; // Needed as this is called inside a function
+
+$stores = cache_config::instance()->get_all_stores();
+$servers = array ();
+
+foreach ($stores as $name => $store) {
+    if (strpos($name, 'memcache') === false) {
+        continue;
+    }
+
+    $storeresult = array();
+    foreach ($store['configuration']['servers'] as $server) {
+        $host = $server[0];
+        if (count($server) > 1) {
+            $port = $server[1];
+        } else {
+            $port = 11211;
+        }
+        $storeresult["$host:$port"] = array('hostname' => $host, 'port' => $port);
+    }
+
+    if (!empty($storeresult)) {
+        $servers[$name] = $storeresult;
+    }
+}
+
 return array (
   'stats_api' => 'Server',
   'slabs_api' => 'Server',
@@ -13,16 +45,6 @@ return array (
   'memory_alert' => '80',
   'hit_rate_alert' => '90',
   'eviction_alert' => '0',
-  'file_path' => 'Temp/',
-  'servers' =>
-  array (
-    'Default' =>
-    array (
-      '127.0.0.1:11211' =>
-      array (
-        'hostname' => '127.0.0.1',
-        'port' => '11211',
-      ),
-    ),
-  ),
+  'file_path' => $PHPMEMCACHEDADMIN_DIR.'/Temp/',
+  'servers' => $servers
 );
\ No newline at end of file
diff --git a/Library/Configuration/Loader.php b/Library/Configuration/Loader.php
index 9e2a91b..03f1e1b 100644
--- a/Library/Configuration/Loader.php
+++ b/Library/Configuration/Loader.php
@@ -26,7 +26,7 @@ class Library_Configuration_Loader
     protected static $_instance = null;
 
     # Configuration file
-    protected static $_iniPath = './Config/Memcache.php';
+    protected static $_iniPath = '/Config/Memcache.php';
 
     # Configuration needed keys
     protected static $_iniKeys = array('stats_api',
@@ -55,6 +55,7 @@ class Library_Configuration_Loader
      */
     protected function __construct()
     {
+        global $CFG;
         # Opening ini file
         self::$_ini = require self::$_iniPath;
     }
diff --git a/View/Header.tpl b/View/Header.tpl
index f489d3d..e4dd12e 100644
--- a/View/Header.tpl
+++ b/View/Header.tpl
@@ -3,8 +3,8 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en">
 <head>
     <title>phpMemcachedAdmin <?php echo CURRENT_VERSION; ?></title>
-    <link rel="stylesheet" type="text/css" href="Public/Styles/Style.css"/>
-    <script type="text/javascript" src="Public/Scripts/Script.js"></script>
+    <link rel="stylesheet" type="text/css" href="<?php echo $PHPMEMCACHEDADMIN_WWW ?>/Public/Styles/Style.css"/>
+    <script type="text/javascript" src="<?php echo $PHPMEMCACHEDADMIN_WWW ?>/Public/Scripts/Script.js"></script>
 </head>
 <body>
 <div style="margin:0pt auto; width:1000px; clear:both;">
